pkg/apis/intevents/v1alpha1/brokercell_lifecycle.go

```diff
@@ -91,10 +91,21 @@ func (bs *BrokerCellStatus) MarkIngressFailed(reason, format string, args ...int
 // to determine if BrokerCellConditionFanout should be marked as true or
 // false.
 func (bs *BrokerCellStatus) PropagateFanoutAvailability(d *appsv1.Deployment) {
-	if duck.DeploymentIsAvailable(&d.Status, false) {
-		brokerCellCondSet.Manage(bs).MarkTrue(BrokerCellConditionFanout)
-	} else {
-		brokerCellCondSet.Manage(bs).MarkFalse(BrokerCellConditionFanout, "DeploymentUnavailable", "Deployment %q is unavailable.", d.Name)
+	deploymentAvailableFound := false
+	for _, cond := range d.Status.Conditions {
+		if cond.Type == appsv1.DeploymentAvailable {
+			deploymentAvailableFound = true
+			if cond.Status == corev1.ConditionTrue {
+				brokerCellCondSet.Manage(bs).MarkTrue(BrokerCellConditionFanout)
+			} else if cond.Status == corev1.ConditionFalse {
```

Comment:
I just did a test, when you create a deployment, initially this is False until all pods are up and running for at least `minReadySeconds`. Therefore I think the behavior is still the same.

Note this is a general problem where clients give up when they see "False", not just for the deployment. I believe we could encounter this for other dependencies as well.

Maybe as a temporary solution we should just use True or Unknown.

Reply:
We should reflect the real status. If that's `False`, we should set status to `False`. CLI should implement the grace period.

Reply:
SG